craft_parts/plugins/base.py

```diff
@@ -36,6 +37,17 @@
     from craft_parts import infos
 
 
+@dataclass(frozen=True)
+class Package:
+    """A dataclass that uniquely identifies a package."""
+
+    name: str
+    version: str
+
+
+PackageFileList = dict[Package, set[pathlib.Path]]
```

Comment:
Perhaps just `PackageFiles` since it's not a list?

Reply:
https://github.com/canonical/craft-parts/pull/931/commits/517e0ede0221c490f6df90106bdff9f6eff9c7bb